android/Samples/KMSample1/build.sh

```diff
@@ -1,61 +1,80 @@
 #!/usr/bin/env bash
-# Build KMSample1
+#
+# Samples: KMsample1
 
-set -e
-set -u
+set -eu
+# set -x: Debugging use, print each statement
+# set -x
 
-display_usage ( ) {
-    echo "build.sh [-no-daemon] [-debug]"
-    echo
-    echo "Build KM Sample 1"
-    echo "  -no-daemon              Don't start the Gradle daemon. Use for CI"
-    echo "  -debug                  Compile only Debug variant"
-    exit 1
-}
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
+. "${THIS_SCRIPT%/*}/../../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
 
-echo Build KMSample1
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+# This script runs from its own folder
+cd "$THIS_SCRIPT_PATH"
+
+################################ Main script ################################
+
+# Definition of global compile constants
+
+CONFIG="release"
+SAMPLE_FLAGS="build"
+
+builder_describe "Build KMSample1 app for Android." \
+  "@../../KMEA" \
+  "clean" \
+  "configure" \
+  "build" \
+  ":app                   KMSample1" \
+  "--ci                   Don't start the Gradle daemon. Use for CI"
+
+# parse before describe_outputs to check debug flags
+builder_parse "$@"
+
+if builder_is_debug_build; then
+  builder_heading "### Debug config ####"
+  CONFIG="debug"
+  SAMPLE_FLAGS="assembleDebug"
+fi
+
+ARTIFACT="app-$CONFIG.apk"
+
+
+builder_describe_outputs \
+  configure             app/libs/keyman-engine.aar \
+  build:app             app/build/outputs/apk/$CONFIG/$ARTIFACT
 
-#
-# Prevents 'clear' on exit of mingw64 bash shell
-#
-SHLVL=0
 
-NO_DAEMON=false
-ONLY_DEBUG=false
 
 # Parse args
-while [[ $# -gt 0 ]] ; do
-    key="$1"
-    case $key in
-        -no-daemon)
-            NO_DAEMON=true
-            ;;
-        -debug)
-            ONLY_DEBUG=true
-            ;;
-        -h|-\?)
-            display_usage
-            ;;
-    esac
-    shift # past argument
-done
-
-echo
-echo "NO_DAEMON: $NO_DAEMON"
-echo "ONLY_DEBUG: $ONLY_DEBUG"
-echo
-
-if [ "$NO_DAEMON" = true ]; then
-  DAEMON_FLAG=--no-daemon
-else
-  DAEMON_FLAG=
+
+if builder_has_option --ci; then
+  SAMPLE_FLAGS="$SAMPLE_FLAGS -no-daemon"
+fi
+
+#### Build action definitions ####
+
+# Check about cleaning artifact paths
+if builder_start_action clean; then
+  rm -rf "$KEYMAN_ROOT/android/Samples/KMSample1/app/build/outputs"
+  builder_finish_action success clean
 fi
 
-if [ "$ONLY_DEBUG" = true ]; then
-  BUILD_FLAG=assembleDebug
-else
-  BUILD_FLAG=build
+if builder_start_action configure; then
+  # Copy Keyman Engine for Android
+  cp "$KEYMAN_ROOT/android/KMEA/app/build/outputs/aar/${CONFIG}/keyman-engine.aar" "$KEYMAN_ROOT/android/Samples/KMSample1/app/libs/keyman-engine.aar"
+
+  builder_finish_action success configure
```

Comment:
```suggestion
  builder_finish_action success configure:app
```

All these just for consistency...